specreduce/utils/utils.py

```diff
@@ -0,0 +1,181 @@
+        will be used. [default: None]
+    pixel: int, list of int, or None
+        Pixel value(s) along the dispersion axis to return cross-dispersion profile.
+        If several specified in list, then the average (method set by `statistic`)
+        profile will be calculated. If None, and `pixel_range` is set, then
+        `pixel_range` will be used. [default: None]
+    pixel_range: tuple of int or None
+        Tuple of (min, max) defining the pixel range (along dispersion axis) to
+        calculate average cross-dispersion profile, up to and not inclusive of max.
+        If None, and `pixel` is not None, `pixel` will be used. If None and `pixel`
+        is also None, this will be interpreted as using the entire dispersion axis
+        to generate an average profile. [default: None]
+    statistic: 'median' or 'mean'
+        If `pixel` specifies multiple pixels, or `pixel_range` is specified, an
+        average profile will be returned. This can be either `median` (default)
+        or `mean`. This is ignored if only one pixel is specified. [default: median]
+    """
+
+    # work on copy of image so input isn't modified
+    image = copy.copy(image)
+
+    # parse image, which will return a spectrum1D (note: this is not ideal,
+    # but will be addressed at some point)
+    parser = _ImageParser()
+    image = parser._parse_image(image, disp_axis = 1)
+
+    # which we then need to make back into a masked array
+    # again this way of parsing the image is not ideal but
+    # thats just how it is for now.
+    image = np.ma.MaskedArray(image.data, mask=image.mask)
+
+    # check inputs
+    if crossdisp_axis != 0 and crossdisp_axis != 1:
+        raise ValueError('`crossdisp_axis` must be 0 or 1.')
+    crossdisp_axis = int(crossdisp_axis)
+    disp_axis = 1 if crossdisp_axis == 0 else 0
+
+    # transpose if disp_axis = 0 just for simplicity of calculations
+    # image is already copied so this won't modify input
+    if disp_axis == 0:
+        image = image.T
+
+    nrows = image.shape[crossdisp_axis]
+    ncols = image.shape[disp_axis]
+
+    if not isinstance(trace, Trace):  # `trace` can be a trace obj
+        if trace is None:  # if None, make a FlatTrace in the center of image
+            trace_pos = nrows / 2
+            trace = FlatTrace(image, trace_pos)
+        elif isinstance(trace, (float, int)):  # if float/int make a FlatTrace
+            trace = FlatTrace(image, trace)
+        else:
+            raise ValueError('`trace` must be Trace object, number to specify '
+                             'the location of a FlatTrace, or None to use center'
+                             ' of image.')
+
+    if statistic not in ['median', 'mean']:
+        raise ValueError("`statistic` must be 'median' or 'mean'.")
+
+    # determine if there is one pixel/wavelength selected or many as either a
+    # list or a tuple to specify a range
+    if pixel is not None:
+        if not (pixel is None) ^ (pixel_range is None):
```

Comment:
can this second line be simplified given that it's within `pixel is not None`?

Reply:
yeah i think so